### PR TITLE
Introducing cachemetrics package

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cache/cachemetrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+			sourceKey,
+		},
+	)
+)
+
+func Register(r prometheus.Registerer) {
+	r.MustRegister(getCounter)
+}
+
+func IncrementRequestCacheCounter(status StatusLabel, source SourceLabel) {
```
Comment (nghialv) [inline pkg/cache/cachemetrics/metrics.go:57]: nit: `Increment` or `Inc`? which is better?

Comment (khanhtc1202) [reply pkg/cache/cachemetrics/metrics.go]: Lets go with `Inc` for short 👍

**Comment by nghialv** [comment]
Thank you.
/lgtm

File: pkg/cache/cachemetrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+const (
+	LabelSourceRedis    SourceLabel = "redis"
+	LabelSourceInmemory SourceLabel = "inmemory"
+)
+
+var (
+	getCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "controlplane_cache_get_operation_total",
```
Comment (nakabonne) [inline pkg/cache/cachemetrics/metrics.go:43]: Is prefix `controlplane` okay? pipecd is better?

But it's okay to determine later.

Comment (khanhtc1202) [reply pkg/cache/cachemetrics/metrics.go:43]: Ops, should be as your suggestion 🙏 lets me address it in this PR 🙏

**Comment by nakabonne** [comment]
I can't wait to see those metrics on our dashboard.
/approve

File: pkg/cache/cachemetrics/metrics.go (hunk 2 above)
Comment (nakabonne) [reply pkg/cache/cachemetrics/metrics.go:43]: Oops, already merged lol

Comment (khanhtc1202) [reply pkg/cache/cachemetrics/metrics.go:43]: @nakabonne plz https://github.com/pipe-cd/pipe/pull/2157 🙏

Comment (nakabonne) [reply pkg/cache/cachemetrics/metrics.go:43]: Gotcha!

**Pull request merged**